### PR TITLE
Spinbox unit prefix

### PR DESCRIPTION
**Pull request by alrik-durand**
I've added a feature to the scientific spinbox to have a default prefix.

## Description
The GUI module can set the _assumed_unit_prefix of a spinbox so that a prefix is assumed if none is provided.

## Motivation and Context
For some module, like confocal scan, unit is almost the same. This let the user use the config file to change a spinbox's behavior if this is supported by the GUI module.

## How Has This Been Tested?
I've checked that this does what it's supposed to on a real setup.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have documented my changes in the changelog (`documentation/changelog.md`)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have checked that the change does not contain obvious errors (syntax, indentation, mutable default values).
- [ ] I have tested my changes using 'Load all modules' on the default dummy configuration with my changes included.
- [ ] All changed Jupyter notebooks have been stripped of their output cells.

### CODE REVIEW

**Comment by Neverhorst** [comment]
I see potential benefit of this PR, however I'm not quite comfortable with the user not being able to enter values without SI-prefix if you set the assumed prefix property.

The discussion should be about if you want to have an SI-prefix when your `value == 0` (this would help operating confocal) or if you want to be able to set a fixed prefix without the ScienDSpinBox rescaling it. (I think this would be the less ambiguous solution and your PR is almost at this point).

If you just want to avoid selection of the entire text (including prefix) inside the SpinBox, you can already get around that by using TAB to select the SpinBox. This will only select the number part of your string without prefix. The only case currently where using TAB will not avoid entering the prefix by hand is when you have zero as value (see previous paragraph).

Long story short: Assuming you use TAB to select the SpinBoxes would it not solve all problems if the zero would maintain the SI prefix?

**Comment by alrik-durand** [comment]
I see your point and indeed this can be a problem in some cases.

This was requested by multiple persons, and I've used this feature for some time and in my case it was quite usefull.

The zero trick could work but I'm worried that when using the slider, the last prefix saved would be nano (in my case) instead of µ.

**Comment by Neverhorst** [comment]
Well I'm not really against it, just thinking if we could make this without excluding a certain value range.

Do you think the other point I mentioned would work? Being able to fix the SI prefix for a ScienDSpinBox?

**Comment by alrik-durand** [comment]
I'm not sure if I understand well what you're proposing. 
The concern I have is : If you set the position to zero by dragging the cursors in the *scan parameter*, and then type a value without prefix, is it going to guess right ?

**Comment by Neverhorst** [comment]
I just checked and you are right. If you drag the slider towards zero, the last number shown has a nano-prefix (or even smaller for longer sliders). So that's not an option.

Another possibility just came to my mind:
What if we just alter your current solution slightly so that it is not checking for a missing unit prefix but instead it checks whether the entered number (as it is, so for example 100) lies within the allowed min-max range. If the number is too small/big it tries to add the default prefix and checks again. If it's still outside the boundaries, it behaves as usual and sets the allowed min or max value instead.
This behavior is more flexible to my mind and does not exclude a certain parameter range. In your particular use case it should also not make a difference.

**Comment by alrik-durand** [comment]
I've implemented your idea
This works on my dummy config
